src/java.base/share/classes/sun/security/x509/X509CRLImpl.java

```diff
@@ -280,13 +280,15 @@ public X500Principal getIssuerX500Principal() {
          *   prevCertIssuer if it does not exist
          */
         private X500Principal getCertIssuer(X509CRLEntryImpl entry,
-                                            X500Principal prevCertIssuer) {
+                X500Principal prevCertIssuer) throws CRLException {
 
             CertificateIssuerExtension ciExt =
                     entry.getCertificateIssuerExtension();
             if (ciExt != null) {
                 GeneralNames names = ciExt.getNames();
-                X500Name issuerDN = (X500Name) names.get(0).getName();
+                if (!(names.get(0).getName() instanceof X500Name issuerDN)) {
+                    throw new CRLException("Parsing error: bad X500 issuer");
```

Comment:
Suggest a slight rewording of the error message: "Parsing error: issuer is not an X.500 DN"